frontend/src/metabase/timelines/components/NewEventModal/NewEventModal.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useCallback, useMemo } from "react";
+import { t } from "ttag";
+import * as Urls from "metabase/lib/urls";
+import Form from "metabase/containers/Form";
+import forms from "metabase/entities/timeline-events/forms";
+import { Collection, Timeline, TimelineEvent } from "metabase-types/api";
+import ModalHeader from "../ModalHeader";
+import { ModalBody } from "./NewEventModal.styled";
+
+export interface NewEventModalProps {
+  timeline?: Timeline;
+  collection: Collection;
+  onSubmit: (
+    values: Partial<TimelineEvent>,
+    collection: Collection,
+    timeline?: Timeline,
+  ) => void;
+  onChangeLocation: (location: string) => void;
+  onClose?: () => void;
+}
+
+const NewEventModal = ({
+  timeline,
+  collection,
+  onSubmit,
+  onChangeLocation,
+  onClose,
+}: NewEventModalProps): JSX.Element => {
+  const initialValues = useMemo(() => {
+    return { timeline_id: timeline?.id };
+  }, [timeline]);
+
+  const handleSubmit = useCallback(
+    async (values: Partial<TimelineEvent>) => {
+      await onSubmit(values, collection, timeline);
+    },
+    [timeline, collection, onSubmit],
+  );
+
+  const handleCancel = useCallback(() => {
+    if (timeline) {
+      onChangeLocation(Urls.timelineInCollection(timeline, collection));
+    } else {
+      onChangeLocation(Urls.timelinesInCollection(collection));
+    }
+  }, [timeline, collection, onChangeLocation]);
```

Comment:
Can we just use `goBack` from `react-router-redux` here or add an `onCancel` prop? This makes it too bound to the collections page, and we'd maybe show this modal in the QB too

Reply:
Fixed